### ISSUE TITLE
Correct statement on number of trees for GBDT

### ISSUE THREAD

**Issue by fritshermans** [INRIA/scikit-learn-mooc#793]
Exercise M6.03 states:

"Both gradient boosting and random forest models improve when increasing the number of trees in the ensemble. However, the scores reach a plateau where adding new trees just makes fitting and scoring slower."

While this statement holds true for random forests, it does not apply to gradient boosting decision trees. In GBDT, adding too many estimators can lead to overfitting. This can be demonstrated by creating a validation curve for GBDT, similar to what is done for random forests (see attached image). The result actually supports the need for hyperparameter tuning or applying early stopping, as shown at the end of exercise M6.03.
![_gbdt_sklearn_mooc](https://github.com/user-attachments/assets/01e6bc02-8035-483c-bbfe-f892b1d390fe)

**Comment by fritshermans** [comment]
I wouldn't mind making a PR with a proposal, if you're fine with that.

**Comment by GaelVaroquaux** [comment]
Yes please! Thanks

**Comment by fritshermans** [comment]
@GaelVaroquaux I prepared a PR. Please let me know if you like the changes.
